validator/delphi_validator/report.py

```diff
@@ -0,0 +1,99 @@
+"""Validation output reports."""
+import sys
+from datetime import date, datetime
+from typing import List, Tuple
+
+class ValidationReport:
+    """Class for reporting the results of validation."""
+    def __init__(self, errors_to_suppress: List[Tuple[str]]):
+        """Initialize a ValidationReport.
+        Parameters
+        ----------
+        errors_to_suppress: List[Tuple[str]]
+            List of error identifications to ignore.
+
+        Attributes
+        ----------
+        errors_to_suppress: List[Tuple[str]]
+            See above
+        num_suppressed: int
+            Number of errors suppressed
+        total_checks: int
+            Number of validation checks performed
+        raised_errors: List[Exception]
+            Errors raised from validation failures
+        raised_warnings: List[Exception]
+            Warnings raised from validation execution
+        suppressed_errors: List[Exception]
+            Errors raised from validation failures not found in `self.errors_to_suppress`
+        """
+        self.errors_to_suppress = errors_to_suppress.copy()
+        self.num_suppressed = 0
+        self.total_checks = 0
+        self.raised_errors = []
+        self.raised_warnings = []
+        self.suppressed_errors = []
+
+    def add_raised_error(self, error):
+        """Add an error to the report.
+        Parameters
+        ----------
+        error: Exception
+            Error raised in validation
+
+        Returns
+        -------
+        None
+        """
+        self.raised_errors.append(error)
+        # Convert any dates in check_data_id to strings for the purpose of comparing
+        # to manually suppressed errors.
+        raised_check_id = tuple([
+            item.strftime("%Y-%m-%d") if isinstance(item, (date, datetime))
+            else item for item in error.check_data_id])
+
+        if raised_check_id not in self.errors_to_suppress:
+            self.suppressed_errors.append(error)
+        else:
+            self.errors_to_suppress.remove(raised_check_id)
+            self.num_suppressed += 1
+
+
+    def increment_total_checks(self):
+        """Records a check."""
+        self.total_checks += 1
+
+    def add_raised_warning(self, warning):
+        """Add a warning to the report.
+        Parameters
+        ----------
+        error: Exception
+            Error raised in validation
+
+        Returns
+        -------
+        None
+        """
+        self.raised_warnings.append(warning)
+
+    def __str__(self):
```

Comment:
Add a test that this formats to what we expect

Reply:
Added tests for the whole report class